libsolidity/analysis/ControlFlowBuilder.cpp

```diff
@@ -310,6 +255,184 @@ bool ControlFlowBuilder::visit(FunctionCall const& _functionCall)
 	return ASTConstVisitor::visit(_functionCall);
 }
 
+bool ControlFlowBuilder::visit(ModifierInvocation const& _modifierInvocation)
+{
+	if (auto arguments = _modifierInvocation.arguments())
+		for (auto& argument: *arguments)
+			appendControlFlow(*argument);
+
+	auto modifierDefinition = dynamic_cast<ModifierDefinition const*>(
+		_modifierInvocation.name()->annotation().referencedDeclaration
+	);
+	if (!modifierDefinition) return false;
+	solAssert(!!modifierDefinition, "");
+	solAssert(!!m_returnNode, "");
+
+	m_placeholderEntry = newLabel();
+	m_placeholderExit = newLabel();
+
+	appendControlFlow(*modifierDefinition);
+	connect(m_currentNode, m_returnNode);
```

Comment:
Does this properly handle `return` in functions with modifiers?

Reply:
Ah yes, it does!

Reply:
Does it have a test that verifies that the order of modifier applications is correct?

Reply:
It should. The modifier flow is added when visiting the modifier invocation and returns to the currently active return node on exit. Its control flow is split between ``m_placeholderEntry`` and ``m_placeholderExit``. After the modifier invocation, ``m_returnNode`` is set to the placeholder exit and control flow continues at the placeholder entry, so a function return should only jump to the placeholder exit. There should be tests for it, though.